environments/environments.py

```diff
@@ -0,0 +1,65 @@
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+class Environment(ABC):
+
+    @abstractmethod
+    def set(self, p, q):
+        """Sets initial conditions for physical system
+
+        Args:
+            p (float): generalized momentum
+            q (float): generalized position
+
+        Raises:
+            NotImplementedError: Class instantiation has no implementation
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def step(self, dt=0.01):
+        """Performs a step in the environment simulator
+
+        Args:
+            dt (float, optional): Time step run for the integration. Defaults to 0.01.
+
+        Raises:
+            NotImplementedError: Class instantiation has no implementation
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def draw(self):
+        """Returns Array of the environment state
+
+        Raises:
+            NotImplementedError: Class instantiation has no implementation
+        """
+        raise NotImplementedError
+
+    def generate_data(self, total_seconds, fps, dt=0.01):
```

Comment:
Add directory argument to store the dataset
e.g: save_dir = None
If save_dir is not none then save the dataset in that directory